You-Can-Code-It/fndr
# Chore update dev

# What this pull request does

## What

- Updated development branch with the main branch
- Fixed some TS issues

## Why

- Luiz had commited some changes on main to deploy, but they were not present on development
- This caused a merge conflict in the package.json
- There were a number of TypeScript types missing (or wrong) which prevented `npm run build`

## How

- I pulled main and dev, merged main into dev and pushed this on a separate branch
- Fixed the issues with `any` in case of Errors, and ts-ignore in the case of Next Auth. Here is a [thread on the types of prisma adapter being screwed up ... ](https://github.com/nextauthjs/next-auth/issues/6106), we'll wait until it is fixed by Next Auth

## Stepping to test this

- commit any changes you have on the banch you are working on 
- fetch this branch using `git fetch --all`
- checkout to `git checkout chore-update-dev`
- run `npm i` and then `npm run build`
- It should build without errors

## Anything else

@luizfiorentino I know I said, deploy from main, but what I should have recommended was: 

- Create a branch from development called `deploy-config-branch`
- On vercel, when it asks which branch you want to deploy from, select `deploy-config-branch`
- That way you can test the deployments without editing main (you should not directly edit main) 
- When the deployment work on your test branch, you PR development and then PR main
- Now on vercel you deploy the main branch instead of your `deploy-config-branch`
